libraries/lib-mixer/Mix.cpp

```diff
@@ -201,15 +200,19 @@ Mixer::NeedsDither(bool needsDither, double rate) const
       // We will call MixVariableRates(), so we need nontrivial resampling
       needsDither = true;
 
-   for (const auto &input : mInputs) {
-      auto &pSequence = input.pSequence;
-      if (!pSequence)
-         continue;
-      auto &sequence = *pSequence;
+   for (const auto &input : mSources) {
+      auto &sequence = input.GetSequence();
+      
       if (sequence.GetRate() != rate)
          // Also leads to MixVariableRates(), needs nontrivial resampling
          needsDither = true;
-      if (mApplyTrackGains) {
+      else if (mApplyGain == ApplyGain::Mixdown &&
```

Comment:
What makes averaging left with right require dither ?

Reply:
I think I know now: because dither is required if there is a multiplication by something other than an integer of the sample values (and that the output format is not float, but surely this must be checked elsewhere), which is the case for mixdown.